### ISSUE TITLE
One condition is shown but another is applied in GenericFilter

### ISSUE THREAD

**Issue by mig-sam** [jmix-framework/jmix#5516]
### Environment

Jmix version: 3.0.999-SNAPSHOT
Jmix Studio Plugin Version: 3.0.1-261
IntelliJ version: IntelliJ IDEA 2026.1

### Bug Description

The user sees a GenericFilter with the condition from the `openOrders` configuration, but the rows are fetched using a different condition -`PropertyCondition.greaterOrEqual("amount", BigDecimal.valueOf(100))`

### Steps To Reproduce

1.Launch demo project
2.Navigate to Scenario 16 view
3.Click button `Set base AFTER activation (amount ≥ 100)`
Following code will be executed:
````
        // Set the base AFTER a configuration is active; it is adopted on the next configuration switch.
        genericFilter.setCurrentConfiguration(genericFilter.getConfiguration("openOrders"));
        genericFilter.getDataLoader().setCondition(PropertyCondition.greaterOrEqual("amount", BigDecimal.valueOf(100)));
        genericFilter.apply();
````

### Current Behavior

One condition is shown but another is applied in GenericFilter

### Expected Behavior

Should be shown condition that is applied

### Sample Project

<img width="1706" height="910" alt="Image" src="https://github.com/user-attachments/assets/4581df2d-279a-45b8-b4eb-1e28958490e9" />

[filter-test.zip](https://github.com/user-attachments/files/30307181/filter-test.zip)

**Comment by knstvk** [comment]
Related to #5351

**Comment by fractal3000** [comment]
## Test scenarios

**Setup**

1. Unzip the attached project, in `build.gradle` set `bomVersion` to the build under test
   (the archive ships with `3.0.999-SNAPSHOT`).
2. Run `./gradlew bootRun`, open `http://localhost:8080`, log in as `admin`/`admin`.
   Demo orders (Open / In progress / Closed, various amounts) are created at startup.

All scenarios show the actual loader condition in a label under the filter — the label and the
grid content must always agree with what the filter displays.

**1. `GenericFilter`: `apply()` after the application replaced the loader condition** (the
reporter's path) — menu: Orders → Scenario 25.

- On open: the configuration *Open Orders* (Status = Open) is active, the grid contains only
  Open orders.
- Click **Replace base (amount >= 100), then apply()**.
- Expected: the label shows `AND[amount greater_or_equal 100, AND[status equal OPEN]]`; the grid
  contains only Open orders with amount ≥ 100.
- Broken behavior (before the fix): the label shows `amount greater_or_equal 100` alone and the
  grid contains In progress / Closed orders while the filter still displays "Status = Open".

**2. `GenericFilter`: the Refresh button** — reopen Scenario 25 (navigate away and back).

- Click **Replace base only**, then press the filter's **Refresh** (apply) button.
- Expected: the same combined condition and grid content as in scenario 1.

**3. `GenericFilter`: a value change after the base was replaced** — reopen Scenario 25.

- Click **Replace base only**, then change the Status condition value to *In progress* in the
  filter itself.
- Expected: `AND[amount greater_or_equal 100, AND[status equal IN_PROGRESS]]`; the grid contains
  only In progress orders with amount ≥ 100.

**4. `GenericFilter`: an operation change after the base was replaced** — reopen Scenario 25.

- Click **Replace base only**, then change the Status condition operation to `<>`.
- Expected: `AND[amount greater_or_equal 100, AND[status not_equal OPEN]]`.

**5. Standalone `GroupFilter`** — menu: Orders → Scenario 26.

- Repeat scenarios 1 and 3 there (the buttons are analogous).
- Expected: the group composes the replaced base with its own conditions the same way.

**Regression**

- Walk over a few other filter views of the project (Scenarios 1–24): switching configurations,
  applying conditions and re-navigation must work as before, no errors in the browser console.

**Covered by automated tests only** (no UI path needed): a configuration whose root group is
built by application code and registered via `GenericFilter.addConfiguration`; the pin that
`DataGridHeaderFilter` is not affected by the change.

**Out of scope of this fix** (documented in the PR): standalone `PropertyFilter`/`JpqlFilter`
attached directly to a loader still load by a replaced base on programmatic paths — addressed
separately in master by PR #5661 behind a property.

[filter-test-5516.zip](https://github.com/user-attachments/files/31912607/filter-test-5516.zip)

**Comment by mig-sam** [comment]
Tested on:
Jmix version: 3.0.999-SNAPSHOT
Jmix Studio Plugin Version: 3.1.SNAPSHOT8083-262
IntelliJ version: IntelliJ IDEA 2026.2.1 RC
